scripts/run_system_test.py

```diff
@@ -270,39 +272,27 @@ def check_training_metric(model_name, model_mgr, eval_dir_mgr, min_train_metric,
                       f"to check for this model. Exiting.")
         sys.exit(-1)
 
-    platform = model_mgr.get_model_platform()
+    model_config = ModelConfig.load(model_mgr.get_model_config())
+    trainer_cls = jb_loader.load_class(model_config.trainer.fqcn)
+    evaluator_cls = jb_loader.load_class(model_config.evaluator.fqcn)
 
     train_metric_name = "accuracy"
     training_metric = training_data.results.accuracy[-1]
 
-    if platform in ['pytorch', 'pytorch_privacy']:
-        eval_metric_name = "balanced_accuracy"
-        eval_metric = eval_data.results.metrics.balanced_accuracy
-
-    elif platform in ['detectron2', 'mmdetection']:
-        eval_metric_name = "mAP"
-        eval_metric = eval_data.results.metrics.bbox['mAP']
-
-    elif platform in ['tensorflow']:
-        eval_metric_name = "accuracy"
-        eval_metric = eval_data.results.metrics.accuracy
-
-    else:
-        logging.error(f"Unknown platform type detected for model {model_name}. Exiting.")
-        sys.exit(-1)
+    eval_metric = evaluator_cls.get_default_metric_value(eval_data)
 
     if training_metric >= min_train_metric:
         logging.info(f"Training {train_metric_name} for {model_name} of {training_metric} exceeded "
                      f"min {train_metric_name}: {min_train_metric}.")
 
         # Met the training threshold, so check the eval metric in the same fashion.
         if eval_metric >= min_eval_metric:
-            logging.info(f"Evaluation {eval_metric_name} for {model_name} of {eval_metric} exceeded "
-                         f"min {eval_metric_name}: {min_eval_metric}.")
+            logging.info(f"Evaluation metric for {model_name} of {eval_metric} exceeded "
+                         f"min metric: {min_eval_metric}.")
             return True
         else:
-            logging.error(f"Evaluation {eval_metric_name} for {model_name} of {eval_metric} did NOT exceed min "
-                          f"{eval_metric_name}: {min_eval_metric}. See evaluation output file for details.")
+            logging.error(f"Evaluation metric for {model_name} of {eval_metric} did NOT exceed min "
+                          f"metric: {min_eval_metric}. See evaluation output file for details.")
```

Comment:
I feel like these changes in line 290-295 are a downgrade that I think we can avoid. Compare the following log lines:

OLD: Evaluation mAP for text_detect/dt2/ut of 3.81807962996225e-05 exceeded min mAP: 3e-05.
PROPOSED CHANGE: Evaluation metric for text_detect/dt2/ut of 3.825903640519213e-05 exceeded min metric: 3.8e-05.

How would you feel if I updated `get_default_metric_value` for each evaluator to return both the value of the metric and its name?

Reply:
Sure.  This is a placeholder for the work that Bill is going to do.  If you want to take the time now, that is fine, but it will probably be rewritten in the next few weeks. (I hope.)